ethereum/contracts/BasicOutboundChannel.sol

```diff
@@ -11,14 +11,15 @@ contract BasicOutboundChannel is OutboundChannel, ChannelAccess, AccessControl {
     // Governance contracts will administer using this role.
     bytes32 public constant CONFIG_UPDATE_ROLE = keccak256("CONFIG_UPDATE_ROLE");
 
-    uint64 public nonce;
+    mapping(address => uint64) public nonces;
 
     // Only messages originating from this account will
     // be allowed through the channel.
     address public principal;
 
     event Message(
         address source,
+        address origin,
```

Comment:
I think we should drop the `origin` name, as it could be misleading for reviewers and auditors. It makes it sound like we're relying on [tx.origin](https://hackernoon.com/hacking-solidity-contracts-using-txorigin-for-authorization-are-vulnerable-to-phishing) which has flaws.

I think we should perhaps just call it account:
```suggestion
        address account,
```